torchtune/modules/vision_transformer.py

```diff
@@ -0,0 +1,445 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Optional, Tuple, Union
+
+import torch
+from torch import nn
+
+from torchtune.modules import Fp32LayerNorm
+from torchtune.modules.transformer import _get_clones
+
+"""
+[tiles, patches and tokens]
+
+We support tile-cropping in our VisionTransformer. This can be a bit hard to understand
+at first glance.
+
+Tiles: Result of a pre-processing transform that breaks the image into tiles,
+so the input to a VisionTransformer can be a stack of tiles, instead of a downsized image.
+E.g. if you have a 1600x1600 image, and your VisionTransformer only accepts images with shape 400x400,
+instead of downsizing it to 400x400, you can tile it into 4 tiles of size 400x400. The transformer
+will still be able to see the whole image if you reshape
+it to (batch_size, num_tiles * num_tokens_per_tile, emb_dim).
+Any image can be generalized to a one-tile image.
+
+Patches: In the VisionTransformer, each tile is divided into patches by a convolution.
+If your tile_size is 400x400, and your patch_size = 40, then each tile will become a grid of 10x10 patches.
+Your total number of patches in one image will be num_tiles * num_patches_per_tile. In the case above,
+4 * 100 = 400 patches.
+
+Tokens: Each patch is flattened by the VisionTransformer. This is now called a token embedding.
+A CLS token is added to each tile, and the tokens are fed to the transformers.
```

Comment:
WHat is CLS?

Reply:
I updated the docstring, but its the first token in the sample we show to the transformers. Its a convenient way of having a token to summarize the information if you wanna do something like classification. You can use this close to classify, instead of doing something like mean(tokens).